test/hardhat.go

```diff
@@ -0,0 +1,84 @@
+package test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type HardhatTest struct {
+	t          *testing.T
+	checkFuncs []func(string, error)
+	setupFuncs []func() error
+	projectDir string
+}
```

Comment:
struct specifically for running `npm hardhat test`. We can add support setup functions on the struct for `hardhat compile`, `verify`, etc in the future if needed for the live test.